packages/lesmis-server/src/routes/TranslatableRoute.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { NextFunction } from 'express';
+import { Route } from '@tupaia/server-boilerplate';
+
+// Overwrite the respond function to include a translate step
+// Use translationKeys to define which pieces of the response body
+// are translated for a route
+
+type StringKey = {
+  type: 'string';
+}
+
+type ObjectKey = {
+  type: 'object';
+  properties: { [key: string]: TranslationKey };
+}
+
+type ArrayKey = {
+  type: 'array';
+  items: TranslationKey;
+  where?: (entry: any) => boolean;
+}
+
+type TranslationKey = StringKey | ObjectKey | ArrayKey;
+type TranslationValue = undefined | string | TranslationValue[] | { [key: string]: TranslationValue };
+
+export class TranslatableRoute extends Route {
+  translationSubGroup: string = '';
```

Comment:
This is clear to me now that I have read this file, but wasn't at all clear when I was reading the route implementations. First impression was that it might be used where a route class responds to multiple api paths (e.g. the collection and the individual record), to specify which of those paths ran through the translation. I'm sitting here trying to think up a name or a different place of putting it that might avoid that, but not coming up with much. Could you make some suggestions? My only thought so far is that the `translationSchema` has a top level field `domain` or `scope`, so it'd look like this for the dashboard route:

```
    this.translationSchema = {
      domain: 'dashboards',
      type: 'array',
      items: {
        type: 'object',
        properties: {
          dashboardName: {
            type: 'string'
          },
        }
      }
    };
```